src/jdk.jfr/share/classes/jdk/jfr/internal/FilePurger.java

```diff
@@ -63,6 +63,13 @@ private static void removeOldest() {
     }
 
     private static boolean delete(SafePath p) {
+        try {
+            if (!SecuritySupport.exists(p)) {
+                return true;
+            }
+        } catch (IOException e) {
```

Comment:
Looks like there could be a race here. If the file is removed after the exists check, but before the file is deleted, the method may return false incorrectly.

Reply:
Yes, well spotted. However, does it matter? If this method return false, it means the file will still be in the list of paths to remove, and upon next purge, the SecuritySupport.exists(p) would see the file doesn't exist, and return true. 

I added it because SecuritySupport.delete() might throw an exception (via Files.delete()) if the file doesn't exists, causing non-existent files to be left in the purge list until removeOldest() removes them.